app/uk/gov/hmrc/gform/dms/DmsSubmissionController.scala

```diff
@@ -67,10 +73,15 @@ class DmsSubmissionController(
           TextExpression(Constant(metadata.customerId)),
           metadata.classificationType,
           metadata.businessArea)
-        _ <- fileUpload.submitEnvelope(submission, summaries, dmsSubmission, 0)
+        _ <- fromFutureA(fileUpload.submitEnvelope(submission, summaries, dmsSubmission, 0))
       } yield {
         NoContent
       }
+
+      result.leftMap { l =>
+        Logger.info(s"Problem submitting to DMS: ${l.error}")
+        InternalServerError
```

Comment:
Hmm, I am a bit worried by this.

Normally such errors are handled in ErrorHandler, according to the upstream error.

I would expect errors from the file upload to be handled in the same way, albeit that you might want to additionally log here.